### PR TITLE
Remove Pod mutating webhook which is no longer used

### LINKED ISSUE #3885: Remove the Pod Mutating Admission webhook
## Feature Request

**Is your feature request related to a problem? Please describe:**
We have scaling and troubleshooting concerns about using Kubernetes webhooks on large clusters. Most of Chaos Mesh's webhook configuration is on Custom Resources which is fine, but we want the ability to remove the one on Pods.

Per [this conversation](https://cloud-native.slack.com/archives/C0193VAV272/p1672898951803299) in Slack, it appears that webhook is no longer used.

**Describe the feature you'd like:**
Remove the `inject-v1-pod` webhook and all references.

**Describe alternatives you've considered:**
If this webhook was still needed in some scenarios, an alternative would be adding a config option in helm to de-activate it.

**Teachability, Documentation, Adoption, Migration Strategy:**
Since this component is strictly internal and we believe it is no longer used, no migration strategy appears neccessary.

### PR DESCRIPTION
**Pull request by raymyers**
### What problem does this PR solve?

Close #3885

### What's changed and how it works?
Remove the inject-v1-pod webhook and all references. We believe it is no longer used, the E2E tests should confirm.

### Related changes

- [ ] This change also requires further updates to the [website](https://github.com/chaos-mesh/website) (e.g. docs)
- [ ] This change also requires further updates to the `UI interface`
- Need to **cheery-pick to release branches**
  - [ ] release-2.5
  - [ ] release-2.4

### Checklist

CHANGELOG

- [X] I have updated the `CHANGELOG.md`
- [ ] I have labeled this PR with "no-need-update-changelog"

Tests

It seems appropriate to run the E2E, I was unable to do that locally but could use help with that if the PR pipeline doesn't do it.

- [X] Unit test
- [ ] E2E test
- [ ] No code
- [ ] Manual test (add steps below)

Side effects

- [ ] **Breaking backward compatibility**

### DCO

If you find the DCO check fails, please run commands like below (Depends on the actual situations. For example, if the failed commit isn't the most recent) to fix it:

```shell
git commit --amend --signoff
git push --force
```

### CODE REVIEW

**Comment by STRRL** [comment]
Hi @raymyers ! Thanks for your contribution! :heart:

File: CHANGELOG.md
Comment (g1eny0ung) [inline CHANGELOG.md:11]: Move this line into the below `### Removed` section.

File: cmd/chaos-controller-manager/main.go
Comment (STRRL) [inline cmd/chaos-controller-manager/main.go:214]: This line could also be removed.

pkg/webhook/config and pkg/webhook/inject could also be removed safely

**Review by STRRL — COMMENTED** [review]
Basically LGTM! Thanks!

There are more codes that could be removed. PTAL.

File: CHANGELOG.md
Comment (raymyers) [reply CHANGELOG.md]: Done, thanks for pointing out.

File: cmd/chaos-controller-manager/main.go
Comment (raymyers) [reply cmd/chaos-controller-manager/main.go]: Done

**Review by g1eny0ung — APPROVED** [review]
@raymyers It's a great first pull request on Chao Mesh 👌🏼. Thank you for working on it! Waiting for @STRRL's approval.

**Comment by hajowieland** [comment]
Thanks for this PR! 
When do we get these changes released with 2.5.1 ?

**Comment by g1eny0ung** [comment]
@STRRL Can we just release a `2.5.1` version?

**Comment by g1eny0ung** [comment]
This change hasn't been merged in the `release-2.5` branch. I'll do it now.

**Comment by g1eny0ung** [comment]
https://github.com/chaos-mesh/chaos-mesh/releases/tag/v2.5.1 Released!

**Pull request merged**
